### PR TITLE
Updated deployment documentation

### PR DESCRIPTION
**Pull request by NickGraham101**
## Context

Continuous deployment has been implemented but the docs have not been updated to reflect that change.

## Changes proposed in this pull request

Updated the deployment docs to reflect the implementation of continuous deployment in a previous commit.

## Guidance to review

Check that the docs clearly and accurately reflect the current process.

## Link to Trello card

https://trello.com/c/5mog82oJ

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Comment by NickGraham101** [comment]
[Update Apply deployment doc](https://trello.com/c/5mog82oJ/500-update-apply-deployment-doc)

**Review by chelnak — APPROVED** [review]
Approved but please update commit message with a full stop.

File: docs/deployment.md [hunk 1]
```diff
@@ -4,9 +4,8 @@ The apply build and release process is split into two separate GitHub Actions pi
 
 - [Build](https://github.com/DFE-Digital/apply-for-teacher-training/actions/workflows/build.yml): This is the main CI pipeline which will automatically trigger a build from a commit to the `main` branch, this pipeline is also run on the merged PR head commit when a pull request is raised targetting `main`. If run from `main` branch this also triggers the deployment pipeline after all tests have succeeded.
 
-- [Deploy](https://github.com/DFE-Digital/apply-for-teacher-training/actions/workflows/deploy.yml): This is the main release pipeline that is used to deploy to all environments. Releases to `production` are triggered manually and the target environments can be chosen prior to deployment.
+- [Deploy](https://github.com/DFE-Digital/apply-for-teacher-training/actions/workflows/deploy.yml): This is the main release pipeline that is used to deploy to all environments. Continuous Deployment is enabled so a successful build on `main` triggers deployments to all environments including `production`.  Deployments can be triggered manually and the target environments can be chosen prior to deployment, see the [Manual Deployment guide](manual-deployment.md) for instructions.
```
Comment (saliceti) [inline docs/deployment.md:7]: Can we mention the review apps? They are essential for CD

Comment (NickGraham101) [reply docs/deployment.md]: I've added an explanation of that process

File: docs/deployment.md [hunk 2]
```diff
@@ -4,9 +4,11 @@ The apply build and release process is split into two separate GitHub Actions pi
...
 
-- [Deploy](https://github.com/DFE-Digital/apply-for-teacher-training/actions/workflows/deploy.yml): This is the main release pipeline that is used to deploy to all environments. Releases to `production` are triggered manually and the target environments can be chosen prior to deployment.
+- [Deploy](https://github.com/DFE-Digital/apply-for-teacher-training/actions/workflows/deploy.yml): This is the main release pipeline that is used to deploy to all environments. 
+    - Pull Requests that have the `deploy` label applied will trigger the deployment of a short lived Review App to it's own self contained environment.  This provides an opportunity to manually review the changes prior to merging them to `main`, once the PR has been merged the Review App and it's environment will be deleted.
```
Comment (saliceti) [inline docs/deployment.md:8]: Isn't is _its own self contained_ and _its environment will be_ ?

Comment (NickGraham101) [reply docs/deployment.md]: fixed

File: docs/deployment.md [hunk 3]
```diff
@@ -4,9 +4,11 @@ The apply build and release process is split into two separate GitHub Actions pi
...
+    - Continuous Deployment is enabled so a successful build on `main` triggers deployments to all environments including `production`.  
+    - Deployments can be triggered manually and the target environments can be chosen prior to deployment, see the [Manual Deployment guide](manual-deployment.md) for instructions.
 
-Go to [the Apply Ops Dashboard](https://apply-ops-dashboard.azurewebsites.net) and find the commit you want to deploy
-and check the diff on GitHub to see if there's anything risky.
+The [Apply Ops Dashboard](https://apply-ops-dashboard.azurewebsites.net) provides an overview of what is currently deployed in each environment.
 
-You also have to make sure that you're deploying only work that is safe to deploy. It should be either behind a feature flag or product reviewed, see the [Manual Deployment guide](manual-deployment.md) for instructions.
+You also have to make sure that you're deploying only work that is safe to deploy, it should be either behind a feature flag or product reviewed.
```
Comment (frankieroberto) [inline docs/deployment.md:14]: Slight copy-edit.

```suggestion
You also have to make sure that you’re only merging changes that are safe to deploy. Changes that are not yet ready for production should be behind a feature flag.
```

File: docs/deployment.md [hunk 4]
```diff
@@ -4,9 +4,11 @@ The apply build and release process is split into two separate GitHub Actions pi
...
 
-- [Deploy](https://github.com/DFE-Digital/apply-for-teacher-training/actions/workflows/deploy.yml): This is the main release pipeline that is used to deploy to all environments. Releases to `production` are triggered manually and the target environments can be chosen prior to deployment.
+- [Deploy](https://github.com/DFE-Digital/apply-for-teacher-training/actions/workflows/deploy.yml): This is the main release pipeline that is used to deploy to all environments. 
+    - Pull Requests that have the `deploy` label applied will trigger the deployment of a short lived Review App to its own self contained environment.  This provides an opportunity to manually review the changes prior to merging them to `main`, once the PR has been merged the Review App and its environment will be deleted.
```
Comment (duncanjbrown) [inline docs/deployment.md:8]: What's "short-lived"? Does it delete itself?

**Review by duncanjbrown — APPROVED** [review]
Looks good — 1 q inline!

File: docs/deployment.md (hunk 4 above)
Comment (NickGraham101) [reply docs/deployment.md:8]: Yes, there's a workflow that runs when the PR is merged that deletes the Review App.  The last sentence is trying to explain that, does it need rewording?

**Pull request merged**